R/TealReportCard.R

```diff
@@ -43,21 +43,36 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
     #' card$get_content()[[1]]$get_content()
     #'
     append_fs = function(fs) {
-      checkmate::assert_list(fs)
-      attr_fs <- attr(fs, "formatted")
-      checkmate::assert_character(attr_fs, null.ok = TRUE)
-
+      # todo: whole class should be moved to teal
+      checkmate::assert_class(fs, "teal_slices")
       if (length(fs) != 0) {
+        states_list <- lapply(fs, function(x) {
+          x_list <- shiny::isolate(shiny::reactiveValuesToList(x))
```

Comment:
```suggestion
          x_list <- as.list(x)
```

Reply:
This package shouldn't depend on `teal.slice` so `as.list` shouldn't be used neither. After we merge it I'm planning to move `teal.reporter::TealReporterCard` to follow a proper dependency direction. 
- `teal` imports `teal.reporter` so `teal` should extend `teal.reporter` features not otherwise.


I'm postponing this move as we will have to open `filter_panel_refactor@main` branch also in `teal.modules.*` and replace `teal.reporter::TealReporterCard` in all modules.

Reply:
All right.